app/src/main/java/com/example/instasnap/Register.java

```diff
@@ -0,0 +1,164 @@
+package com.example.instasnap;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthInvalidCredentialsException;
+import com.google.firebase.auth.FirebaseAuthUserCollisionException;
+import com.google.firebase.auth.FirebaseAuthWeakPasswordException;
+import com.google.firebase.auth.FirebaseUser;
+
+public class Register extends AppCompatActivity {
+
+    private EditText _editTextEmail;
+    private EditText _editTextPassword;
+    private Button _registerButton;
+    private FirebaseAuth _mAuth;
+    private ProgressBar _registerProgressBar;
+    private TextView _loginHereTextView;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_register);
+
+        initialize();
+
+        setLoginNowButtonListener();
+
+        setRegisterButtonListener();
+
+    }
+
+    private void initialize() {
+        // Initializing registration objects
+        _mAuth = FirebaseAuth.getInstance();
+        _editTextEmail = findViewById(R.id.register_email_text_input);
+        _editTextPassword = findViewById(R.id.register_password_text_input);
+        _registerButton = findViewById(R.id.register_button);
+        _registerProgressBar = findViewById(R.id.register_progressBar);
+        _loginHereTextView = findViewById(R.id.login_now);
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        // Check if user is signed in (non-null) and update UI accordingly.
+        FirebaseUser currentUser = _mAuth.getCurrentUser();
+        if(currentUser != null){
+            Intent homepageIntent = new Intent(getApplicationContext(), HomePage.class);
+            startActivity(homepageIntent);
+            finish();
+        }
+    }
+
+    @Override
+    public void onBackPressed() {
+        Intent loginIntent = new Intent(getApplicationContext(), Login.class);
+        startActivity(loginIntent);
+        finish();
+    }
+
+    private void setLoginNowButtonListener() {
```

Comment:
Is this suppose to be the "Back to log-in" ?

Reply:
Write now it navigates to the login screen, but since the user is considered logged in ince the registeratiin phase is done, it jumps to the homepage immedietly

Reply:
if the user is logged in we should not be able to get here

Reply:
Thats exactly what i am saying, if the user is logged in, he will be navigated to HomePage class when onStart() in Register class is ruuning, he will indicate that there is a logged in user and will create a HomePage intent